airflow/providers/amazon/aws/sensors/s3_key.py

```diff
@@ -106,3 +107,86 @@ def get_hook(self) -> S3Hook:
+        or relative path from root level. When it's specified as a full s3://
+        url, please leave bucket_name as `None`.
+    :type bucket_key: str
+    :param bucket_name: Name of the S3 bucket. Only needed when ``bucket_key``
+        is not provided as a full s3:// url.
+    :type bucket_name: str
+    :param wildcard_match: whether the bucket_key should be interpreted as a
+        Unix wildcard pattern
+    :type wildcard_match: bool
+    :param aws_conn_id: a reference to the s3 connection
+    :type aws_conn_id: str
+    :param verify: Whether or not to verify SSL certificates for S3 connection.
+        By default SSL certificates are verified.
+        You can provide the following values:
+
+        - ``False``: do not validate SSL certificates. SSL will still be used
+                 (unless use_ssl is False), but SSL certificates will not be
+                 verified.
+        - ``path/to/cert/bundle.pem``: A filename of the CA cert bundle to uses.
+                 You can specify this argument if you want to use a different
+                 CA cert bundle than the one used by botocore.
+    :type verify: bool or str
+    :param summarizer_fn: function that receives the list of the S3 objects,
+        and returns the boolean.
+    :type summarizer_fn: Optional[Callable[..., bool]]
+
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        summarizer_fn: Optional[Callable[..., bool]] = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.summarizer_fn_user = summarizer_fn
+
+    def poke(self, context):
+        if super().poke(context=context) is False:
+            return False
+
+        s3_objects = self.get_files(s3_hook=self.get_hook())
+        summarizer_fn = self.summarizer_fn_user
+        if summarizer_fn is None:
+            summarizer_fn = self.summarizer_fn
+        return summarizer_fn(s3_objects)
+
+    def get_files(self, s3_hook: S3Hook) -> List:
+        """Gets a list of files in the bucket"""
+        prefix = self.bucket_key
+        delimiter = '/'
+        config = {
+            'PageSize': None,
+            'MaxItems': None,
+        }
+        if self.wildcard_match:
+            prefix = re.split(r'[*]', self.bucket_key, 1)[0]
+
+        paginator = s3_hook.get_conn().get_paginator('list_objects_v2')
+        response = paginator.paginate(
+            Bucket=self.bucket_name, Prefix=prefix, Delimiter=delimiter, PaginationConfig=config
+        )
+        keys = []
+        for page in response:
+            if 'Contents' in page:
+                _temp = [k for k in page['Contents'] if isinstance(k.get('Size', None), (int, float))]
+                keys = keys + _temp
+        return keys
+
+    def summarizer_fn(self, data: List) -> bool:
+        """Default function for checking that S3 Objects have size more than 0"""
+        return sum([f.get('Size', 0) for f in data if isinstance(f, dict)]) > 0
```

Comment:
This would return true if _any_ files have content, not all, which I would expect.

This should possibly be

```suggestion
        return all(f.get('Size', 0) > 0 for f in data if isinstance(f, dict))
```

Reply:
Why do we even need the `isinstance` check here -- `get_files` already filters this.

Reply:
@ashb You are right